code/modules/hazmat/crystal_processor.dm

```diff
@@ -0,0 +1,55 @@
+#define CRYSTAL_MACHINE_COLOR "colorizer"
+#define CRYSTAL_MACHINE_SHAPE "shaper"
+#define CRYSTAL_MACHINE_SIZE "sizer"
+/obj/machinery/hazmat/crystal_processor
+	name = "crystal processor"
+	desc = "Modify whatever crystal you got."
+	icon_state = "crystal_maker"
+	var/machinery_type = CRYSTAL_MACHINE_COLOR
+	var/processing_speech = "Doing the Needful to CRYSTAL"
+
+/obj/machinery/hazmat/crystal_maker/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/crystal))
+		var/obj/item/crystal/C = I
+		var/obj/item/crystal/newcrystal = new(get_turf(src))
+		C.malf_chance += 5
+		newcrystal.malf_chance = C.malf_chance
+		switch(machinery_type)
```

Comment:
turn this into a proc instead and have the subtypes override it, it makes no sense for this to be a switch goof you know that.